minidump-processor/src/op_analysis.rs

```diff
@@ -378,69 +629,218 @@ mod amd64 {
+                Opcode::JO
+                | Opcode::JNO
+                | Opcode::JB
+                | Opcode::JNB
+                | Opcode::JZ
+                | Opcode::JNZ
+                | Opcode::JA
+                | Opcode::JNA
+                | Opcode::JS
+                | Opcode::JNS
+                | Opcode::JP
+                | Opcode::JNP
+                | Opcode::JL
+                | Opcode::JGE
+                | Opcode::JG
+                | Opcode::JLE => return Ok(None),
+
+                _ => return Ok(Some(InstructionPointerUpdate::NoUpdate)),
             }
+            Ok(None)
+        }
+    }
 
-            Ok(())
+    /// A subset of opcodes that we support for deriving precise memory access behaviour
+    /// They are either commonly seen in crashes,
+    /// or known to appear in specific inconsistent crashes
+    #[derive(Copy, Clone)]
+    #[allow(clippy::upper_case_acronyms)]
+    enum AccessDerivableOpcode {
+        ADD,
+        CALL,
+        CMP,
+        DEC,
+        INC,
+        JMP,
+        JMPF,
+        JO,
+        JNO,
+        JB,
+        JNB,
+        JZ,
+        JNZ,
+        JA,
+        JNA,
+        JS,
+        JNS,
+        JP,
+        JNP,
+        JL,
+        JGE,
+        JG,
+        JLE,
+        LEA,
+        MOV,
+        MOVAPS,
+        MOVUPS,
+        POP,
+        PUSH,
+        RETF,
+        RETURN,
+        SUB,
+        /// See https://bugzilla.mozilla.org/show_bug.cgi?id=1831370
+        UCOMISS,
+    }
+
+    impl AccessDerivableOpcode {
+        fn from_opcode(opcode: Opcode) -> Option<Self> {
+            macro_rules! convert {
+                ( $($name:ident),* ) => {
+                    match opcode {
+                        $(Opcode::$name => Some(Self::$name),)*
+                        _ => None
+                    }
+                }
+            }
+            convert![
+                ADD, CALL, CMP, DEC, INC, JMP, JMPF, JO, JNO, JB, JNB, JZ, JNZ, JA, JNA, JS, JNS,
+                JP, JNP, JL, JGE, JG, JLE, LEA, MOV, MOVAPS, MOVUPS, POP, PUSH, RETF, RETURN, SUB,
+                UCOMISS
+            ]
         }
+    }
+
+    #[derive(Default)]
+    struct MemoryOperandInfo {
+        pub base_reg: Option<RegSpec>,
+        pub index_reg: Option<RegSpec>,
+        pub scale: Option<u8>,
+        // Reg operands disp are i32, Displacement operands are i64
+        pub disp: Option<i64>,
+    }
 
-        fn calculate_address(
-            &self,
-            access_size: Option<u8>,
-            register_operand_info: RegOperandInfo,
-        ) -> Result<MemoryAccess, OpAnalysisError> {
-            let mut access = MemoryAccess {
+    impl MemoryOperandInfo {
+        pub fn try_from_operand(op: Operand) -> Option<Self> {
+            let mut info = MemoryOperandInfo::default();
+            match op {
+                Operand::DisplacementU32(disp) => info.disp = Some(disp as i32 as i64),
```

Comment:
Note: This results in a different behaviour from the current `Rust-minidump`. 
Referring to [Bug 1909661](https://bugzilla.mozilla.org/show_bug.cgi?id=1909661), `yaxpeax` gives the displacement as a `u32`, but the CPU will treat the displacement as a signed number. It is debatable whether `yaxpeax`'s interpretation is correct. Currently `Rust-minidump` treats the displacement as a `u32` and cast it to a `u64` address (ie. no sign extension), resulting in a different address than the one actually being accessed.
Since this features require the memory address to be accurate, this implementation sign extends the displacement by casting the `u32` to an `i32` before then casting to `i64` to obtain the actual address accessed.